web/hooks/use-group-supabase.ts

```diff
@@ -277,3 +279,29 @@ export function useGroupsWhereUserHasRole(userId: string | undefined) {
   }, [userId])
   return groupsAndRoles
 }
+
+export function useMemberPrivateGroups(userId: string | undefined) {
+  const [memberPrivateGroups, setMemberPrivateGroups] = useState<
+    Group[] | undefined
+  >(undefined)
+  useEffect(() => {
+    if (userId) {
+      getMemberPrivateGroups(userId).then((result) => {
+        setMemberPrivateGroups(result)
+      })
+    }
+  }, [userId])
+  return memberPrivateGroups
+}
+
+export function useYourNonPrivateNonModeratorGroups(userId: string) {
+  const [groups, setGroups] = useState<Group[] | undefined>(undefined)
```

Comment:
Should be able to use `usePersistentInMemoryState` instead of `useState` for better UX on returning to the page